pytest_fixtures/component/leapp_client.py

```diff
@@ -0,0 +1,204 @@
+    },
+    'rhel8_aps': {
+        'id': 'rhel-8-for-x86_64-appstream-rpms',
+        'name': f'Red Hat Enterprise Linux 8 for x86_64 - AppStream RPMs {RHEL8_VER}',
+        'releasever': RHEL8_VER,
+        'reposet': 'Red Hat Enterprise Linux 8 for x86_64 - AppStream (RPMs)',
+    },
+    'rhel9_bos': {
+        'id': 'rhel-9-for-x86_64-baseos-rpms',
+        'name': f'Red Hat Enterprise Linux 9 for x86_64 - BaseOS RPMs {RHEL9_VER}',
+        'releasever': RHEL9_VER,
+        'reposet': 'Red Hat Enterprise Linux 9 for x86_64 - BaseOS (RPMs)',
+    },
+    'rhel9_aps': {
+        'id': 'rhel-9-for-x86_64-appstream-rpms',
+        'name': f'Red Hat Enterprise Linux 9 for x86_64 - AppStream RPMs {RHEL9_VER}',
+        'releasever': RHEL9_VER,
+        'reposet': 'Red Hat Enterprise Linux 9 for x86_64 - AppStream (RPMs)',
+    },
+}
+
+
+@pytest.fixture(scope='module')
+def module_stash(request):
+    """Module scoped stash for storing data between tests"""
+    # Please refer the documentation for more details on stash
+    # https://docs.pytest.org/en/latest/reference/reference.html#stash
+    request.node.stash[synced_repos] = {}
+    return request.node.stash
+
+
+@pytest.fixture(scope='module')
+def module_leapp_lce(module_target_sat, module_sca_manifest_org):
+    return module_target_sat.api.LifecycleEnvironment(organization=module_sca_manifest_org).create()
+
+
+@pytest.fixture
+def function_leapp_cv(module_target_sat, module_sca_manifest_org, leapp_repos, module_leapp_lce):
+    function_leapp_cv = module_target_sat.api.ContentView(
+        organization=module_sca_manifest_org
+    ).create()
+    function_leapp_cv.repository = leapp_repos
+    function_leapp_cv = function_leapp_cv.update(['repository'])
+    function_leapp_cv.publish()
+    cvv = function_leapp_cv.read().version[0]
+    cvv.promote(data={'environment_ids': module_leapp_lce.id, 'force': True})
+    return function_leapp_cv.read()
+
+
+@pytest.fixture
+def function_leapp_ak(
+    module_target_sat,
+    function_leapp_cv,
+    module_leapp_lce,
+    module_sca_manifest_org,
+):
+    return module_target_sat.api.ActivationKey(
+        content_view=function_leapp_cv,
+        environment=module_leapp_lce,
+        organization=module_sca_manifest_org,
+    ).create()
+
+
+@pytest.fixture
+def leapp_repos(
+    default_architecture,
+    module_stash,
+    upgrade_path,
+    module_target_sat,
+    module_sca_manifest_org,
+):
+    """Enable and sync RHEL BaseOS, AppStream repositories"""
+    source = upgrade_path['source_version']
+    target = upgrade_path['target_version']
+    all_repos = []
+    for rh_repo_key in RHEL_REPOS:
+        release_version = RHEL_REPOS[rh_repo_key]['releasever']
+        if release_version in str(source) or release_version in target:
+            prod = 'rhel' if 'rhel7' in rh_repo_key else rh_repo_key.split('_')[0]
+            if module_stash[synced_repos].get(rh_repo_key, None):
```

Comment:
```suggestion
            if module_stash[synced_repos].get(rh_repo_key, False):
```

Reply:
Initially module_stash[synced_repos] is None so we are checking if we don't have cached repository.